pkg/controller/interconnect/interconnect_controller.go

```diff
@@ -719,13 +711,19 @@ func ensureCAIssuer(r *ReconcileInterconnect, instance *v1alpha1.Interconnect, n
 	return nil
 }
 
-// getPodNames returns the pod names of the array of pods passed in
-func getPodNames(pods []corev1.Pod) []string {
-	var podNames []string
-	for _, pod := range pods {
-		if pod.GetObjectMeta().GetDeletionTimestamp() == nil {
-			podNames = append(podNames, pod.Name)
+func getPodStatus(r *ReconcileInterconnect, instance *v1alpha1.Interconnect) olm.DeploymentStatus {
+	// List the pods for this deployment
+	var status olm.DeploymentStatus
+	depFound := &appsv1.Deployment{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, depFound)
```

Comment:
It might be cleaner to use the placement value to determine which method to call for the pod statuses?

Reply:
changed